pyanaconda/ui/gui/spokes/blivet_gui.py

```diff
@@ -45,6 +49,66 @@
 # export only the spoke, no helper functions, classes or constants
 __all__ = ["BlivetGuiSpoke"]
 
+
+class BlivetGUIAnacondaClient(BlivetGUIClient):
+    """"The request handler for the Blivet-GUI."""
+
+    def __init__(self):  # pylint: disable=super-init-not-called
+        self.mutex = Lock()
+        self._proxy = None
+        self._result = None
+
+    def initialize(self):
+        """Initialize the client."""
+        self._proxy = STORAGE.get_proxy(BLIVET_PARTITIONING)
+
+    def _send(self, data):
+        """Send a message to a server."""
+        self._result = bytes(self._proxy.SendRequest(data))
+
+    def _recv_msg(self):
+        """Receive a message from server."""
+        return self._result
+
+    def configure_storage(self):
+        """Configure the storage."""
+        task_path = self._proxy.ConfigureWithTask()
+        task_proxy = STORAGE.get_proxy(task_path)
+        sync_run_task(task_proxy)
+
+    def validate_storage(self):
+        """Validate the storage."""
+        task_path = self._proxy.ValidateWithTask()
+        task_proxy = STORAGE.get_proxy(task_path)
+        sync_run_task(task_proxy)
+
+    def apply_storage(self):
+        """Apply the changes."""
+        storage_proxy = STORAGE.get_proxy()
+        storage_proxy.ApplyPartitioning(BLIVET_PARTITIONING.object_path)
+
+    def get_actions(self):
+        """Get the current actions."""
+        # FIXME: Add DBus support for this.
+        # return self._storage_playground.devicetree.actions.find()
+        return []
+
+    def sort_actions(self):
+        """Prune and sort the actions."""
+        # FIXME: Shouldn't we always do this?
+        # self._storage_playground.devicetree.actions.prune()
+        # self._storage_playground.devicetree.actions.sort()
+        pass
+
+    def set_new_swaps(self):
+        """Set new swaps."""
+        # FIXME: Add DBus support for this.
```

Comment:
You'll need this for custom spoke too -- https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/ui/gui/spokes/custom_storage.py#L226